### PR TITLE
fix: Fixes issue with thirdparty passwordless user in dashboard

### PR DESCRIPTION
**Pull request by sattvikc**
## Summary of change

(A few sentences about this PR)

## Related issues

-   https://github.com/supertokens/supertokens-python/issues/472

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates

-   [ ] Changelog has been updated
-   [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `supertokens_python/constants.py`
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [ ] Changes to the version if needed
    -   In `setup.py`
    -   In `supertokens_python/constants.py`
-   [ ] Had installed and ran the pre-commit hook
-   [ ] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.
-   [ ] If have added a new web framework, update the `supertokens_python/utils.py` file to include that in the `FRAMEWORKS` variable
-   [ ] If added a new recipe that has a User type with extra info, then be sure to change the User type in supertokens_python/types.py
-   [ ] Make sure that `syncio` / `asyncio` functions are consistent.
-   [ ] If access token structure has changed
    -   Modified test in `tests/sessions/test_access_token_version.py` to account for any new claims that are optional or omitted by the core

## Remaining TODOs for this PR

-   [ ] Item1
-   [ ] Item2

### CODE REVIEW

File: supertokens_python/recipe/dashboard/utils.py [hunk 1]
```diff
@@ -309,58 +309,40 @@ async def get_user_for_recipe_id(
...
 
     elif recipe_id == PasswordlessRecipe.recipe_id:
         await update_user_dict(
-            pless_get_user_by_id,
-            tppless_get_user_by_id,
-            "passwordless",
-            "thirdpartypasswordless",
+            [pless_get_user_by_id, tppless_get_user_by_id],
+            ["passwordless", "thirdpartypasswordless"],
         )
```
Comment (rishabhpoddar) [inline supertokens_python/recipe/dashboard/utils.py:346]: add proper tests for this API now

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: supertokens_python/recipe/dashboard/utils.py (hunk 1 above)
Comment (sattvikc) [reply supertokens_python/recipe/dashboard/utils.py:346]: Done

File: tests/dashboard/test_dashboard.py [hunk 2]
```diff
@@ -203,3 +211,60 @@ async def test_that_first_connection_uri_is_selected_among_multiple_uris(
...
+        },
+    )
+    res_json = res.json()
+    assert res_json["status"] == "NO_USER_FOUND_ERROR"
+
+    res = app.get(
+        url="/auth/dashboard/api/user",
+        params={
+            "userId": pluser.user.user_id,
+            "recipeId": "thirdparty",
```
Comment (rishabhpoddar) [inline tests/dashboard/test_dashboard.py:265]: how is this combination recipe? if you pass recipeId as thirdparty?

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,6 +8,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [unreleased]
 
+## [0.18.8] - 2024-01-23
```
Comment (rishabhpoddar) [inline CHANGELOG.md:11]: version 0.18.7 has not been released yet.. please check properly next time

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (sattvikc) [reply CHANGELOG.md]: Fixed

**Review by rishabhpoddar — APPROVED** [review]

**Pull request merged**